internal/cluster/check_selection_kvstore.go

```diff
@@ -0,0 +1,48 @@
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/trento-project/trento/internal/consul"
+)
+
+func GetCheckSelection(client consul.Client, clusterId string) (string, error) {
+	err := client.WaitLock(consul.KvClustersPath)
+	if err != nil {
+		return "", errors.Wrap(err, "error waiting for the lock for clusters")
```

Comment:
Empty strings? 😏

Reply:
What should I return?
It we have an error the error message will come from the `err` parameter.
If you are concerned about the usage should be like this: https://github.com/trento-project/trento/blob/cbd750177af363763d8b7aee7b4d0d9606d8c857/agent/discovery/cloud.go#L37

I don't think it is needed. The error message is already coming in the error in your case, so it is actually duplicated

Reply:
This is pretty idiomatic since the empty string is the blank value of the string type, if the err is not nil the string is going to be ignored from the code that handles the error